src/neo/Ledger/Blockchain.cs

```diff
@@ -292,76 +294,28 @@ private RelayResultReason OnNewBlock(Block block)
         {
             if (block.Index <= Height)
                 return RelayResultReason.AlreadyExists;
-            if (block_cache.ContainsKey(block.Hash))
-                return RelayResultReason.AlreadyExists;
-            if (block.Index - 1 >= header_index.Count)
+            if (block.Index - 1 > Height)
             {
                 AddUnverifiedBlockToCache(block);
                 return RelayResultReason.UnableToVerify;
             }
-            if (block.Index == header_index.Count)
-            {
-                if (!block.Verify(currentSnapshot))
-                    return RelayResultReason.Invalid;
-            }
-            else
-            {
-                if (!block.Hash.Equals(header_index[(int)block.Index]))
-                    return RelayResultReason.Invalid;
-            }
             if (block.Index == Height + 1)
             {
-                Block block_persist = block;
-                List<Block> blocksToPersistList = new List<Block>();
-                while (true)
-                {
-                    blocksToPersistList.Add(block_persist);
-                    if (block_persist.Index + 1 >= header_index.Count) break;
-                    UInt256 hash = header_index[(int)block_persist.Index + 1];
-                    if (!block_cache.TryGetValue(hash, out block_persist)) break;
-                }
-
-                int blocksPersisted = 0;
-                foreach (Block blockToPersist in blocksToPersistList)
+                if (!block.Verify(currentSnapshot))
                 {
-                    block_cache_unverified.Remove(blockToPersist.Index);
-                    Persist(blockToPersist);
-
-                    // 15000 is the default among of seconds per block, while MilliSecondsPerBlock is the current
-                    uint extraBlocks = (15000 - MillisecondsPerBlock) / 1000;
-
-                    if (blocksPersisted++ < blocksToPersistList.Count - (2 + Math.Max(0, extraBlocks))) continue;
```

Comment:
@ShawnYun, this extra relaying will be removed, right?
Perhaps this will not be needed anymore, all right @jsolman ?

Reply:
Now we don't have the `blocksToPersistList`. Perhaps we should discuss this question.

Reply:
This was something that surely increase network overall communication on the P2P.

I think it is good to remove but to keep this logic in mind in case we notice problems on syncing.